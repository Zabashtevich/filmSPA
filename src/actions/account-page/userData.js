import { DATA_FETCHED_SUCCESS } from "./constants";

export const dataIsFetched = (ratedMovies, userLists, favoriteMovies) => {
  console.log("hehe");
  return {
    type: DATA_FETCHED_SUCCESS,
  };
};
