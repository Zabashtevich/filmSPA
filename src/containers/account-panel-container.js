import React from "react";
import { useParams } from "react-router-dom";

import { AccountPanel } from "../components";
import {
  FilterContainer,
  RowListItemContainer,
} from "./auxillary/auxillary-containers";

export default function AccountPanelContainer() {
  const params = useParams();
  return (
    <AccountPanel>
      <FilterContainer slug={params.slug} />
      <RowListItemContainer />
    </AccountPanel>
  );
}
