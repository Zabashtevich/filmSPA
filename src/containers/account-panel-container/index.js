import React, { useState, useEffect } from "react";
import { useLocation, useParams } from "react-router-dom";
import { parse } from "query-string";

import { AccountPanel } from "../../components";
import useAuthListener from "../../hooks/useAuthListener";
import useFirestore from "../../hooks/useFirestore";
import { FilterContainer } from "./auxillary";
import { RowListItemContainer } from "../";
import LoadMoreContainer from "../load-more-container";

export default function AccountPanelContainer() {
  const params = useParams();
  const [user] = useAuthListener();
  const { search } = useLocation();
  const [itemsCount, setItemsCount] = useState(10);
  const [parsedSearch, setParsedSearch] = useState(null);

  const [userDataLoading, userData] = useFirestore(
    user && `${user.displayName}`,
    `moviesrated`,
  );
  useEffect(() => {
    console.log(parse(search));
  }, [search]);
  console.log(userData);
  return (
    <AccountPanel>
      <FilterContainer slug={params.slug} />
      <AccountPanel.CardsContainer>
        {!userDataLoading &&
          userData.map((item, index) => (
            <RowListItemContainer
              key={item.id}
              item={item}
              index={index}
              user={user}
              userData={userData}
              accountPanelRow={true}
            />
          ))}
        <LoadMoreContainer
          setItemsCount={setItemsCount}
          visible={userData.length > itemsCount}
          offset={10}
        />
      </AccountPanel.CardsContainer>
    </AccountPanel>
  );
}
