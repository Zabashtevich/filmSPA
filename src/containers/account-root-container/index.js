import React from "react";
import { useSelector } from "react-redux";

import { AccountRoot } from "../../components";
import useAuthListener from "../../hooks/useAuthListener";
import { UtilityModalContainer } from "../";
import {
  UserlistContainer,
  UserlistPlaceholder,
  UserlistCreatelist,
} from "./auxillary";

export default function AccountRootContainer() {
  const [user, userLoading] = useAuthListener();

  const userData = useSelector(({ userData }) => userData);
  const { userlists, loading } = userData;
  return (
    <>
      <UtilityModalContainer />
      <AccountRoot>
        {!userLoading && (
          <AccountRoot.PosterColumn>
            <AccountRoot.Avatar src={user.photoURL} />
            <AccountRoot.Nickname>{user.displayName}</AccountRoot.Nickname>
            <AccountRoot.Link to={`${user.displayName}/edit`}>
              Edit profile
            </AccountRoot.Link>
          </AccountRoot.PosterColumn>
        )}
        <AccountRoot.ContentColumn>
          <AccountRoot.Title>YOUR PROFILE ACTIVITY</AccountRoot.Title>
          <AccountRoot.Subtitle>your lists:</AccountRoot.Subtitle>
          <AccountRoot.Lists>
            <div style={{ display: "flex" }}>
              {!loading &&
                userlists.length > 0 &&
                userlists.map((item) => (
                  <UserlistContainer item={item} key={item.id} />
                ))}
              {!loading && userlists.length < 5 && <UserlistCreatelist />}
              {!loading && userlists.length === 0 && <UserlistPlaceholder />}
            </div>
          </AccountRoot.Lists>
        </AccountRoot.ContentColumn>
      </AccountRoot>
    </>
  );
}
