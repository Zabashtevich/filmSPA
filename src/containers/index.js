export { default as AuthContainer } from "./auth";
export { default as FooterContainer } from "./footer";
export { default as ModalContainer } from "./modal";
export { default as DetailsPosterContainer } from "./details-poster";
export { default as ActorPosterContainer } from "./actor-poster";
export { default as MovieListContainer } from "./movie-list";
export { default as PaginationContainer } from "./pagination";
export { default as DetailsPanelContainer } from "./details-panel";
export { default as HeaderContainer } from "./header";
export { default as PopularContainer } from "./popular";
export { default as MainContainer } from "./main";
export { default as TabsContainer } from "./tabs";
export { default as TrendingContainer } from "./trending";
export { default as SliderContainer } from "./slider";
export { default as AccountContainer } from "./account";
export { default as FilterContainer } from "./filter";
export { default as CreditsContainer } from "./credits";
export { default as NoticeContainer } from "./notice";
export { default as MediaContainer } from "./media";
export { default as GalleryContainer } from "./gallery";
export { default as PopupContainer } from "./popup";
