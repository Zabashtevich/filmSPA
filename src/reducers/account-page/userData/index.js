const initialState = { value: null };

const userData = (state = initialState, action) => {
  switch (action.type) {
    default:
      return state;
  }
};

export default userData;
