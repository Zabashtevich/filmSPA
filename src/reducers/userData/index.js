import {
  FETCH_STARTED,
  FETCH_FINISHED_SUCCESS,
  SET_USER_LISTS,
  SET_FAVORITED_MOVIES,
  SET_RATED_MOVIES,
} from "./constants";

const initialState = {
  loading: true,
  userLists: null,
  favoritedMovies: null,
  ratedMovies: null,
};

const userData = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_STARTED:
      return { ...state, loading: true };
    case FETCH_FINISHED_SUCCESS:
      return { ...state, loading: false };
    case SET_USER_LISTS:
      return { ...state, userLists: action.payload };
    case SET_FAVORITED_MOVIES:
      return { ...state, favoritedMovies: action.payload };
    case SET_RATED_MOVIES:
      return { ...state, ratedMovies: action.payload };
    default:
      return state;
  }
};

export default userData;
