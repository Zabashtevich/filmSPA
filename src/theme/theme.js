export default {
  fontStandart: {
    standart: "'Source Sans Pro', sans-serif",
  },
  fontCardSlider: {
    standart: "'Oswald', sans-serif",
  },
  fontWeightStandart: {
    light: "300",
    normal: "400",
    bold: "600",
  },
  fontWeightCard: {
    light: "200",
    normal: "400",
    bold: "700",
  },
  fontSize: {
    small: "1rem",
    normal: "1.3rem",
    large: "1.5rem",
    huge: "2.5rem",
    logo: "3.5rem",
  },
  colors: {
    header: "#032541",
    secondary: "#1f1f1f",
    inputColor: "gray",
    appMainColor: "#f4f4f4",
    headerInputIcon: "#fff",
  },
};
