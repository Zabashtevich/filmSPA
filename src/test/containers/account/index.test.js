import { AccountContainer } from "../../../containers";
import userProfile from "../../../reducers/user-profile";
import userData from "../../../reducers/user-data";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { combineReducers, createStore } from "redux";
import { ThemeProvider } from "styled-components";
import theme from "../../../theme/theme";

export default function renderWithRedux(
  component,
  {
    initialState,
    store = createStore(
      combineReducers({ userProfile, userData }),
      initialState,
    ),
  } = {},
) {
  return {
    ...render(
      <ThemeProvider theme={theme}>
        <Provider store={store}>{component}</Provider>
      </ThemeProvider>,
    ),
    store,
  };
}

describe("Account container", () => {
  it("render correctly while loading", () => {
    const { getByTestId, queryByTestId, queryByAltText } = renderWithRedux(
      <AccountContainer />,
      {
        initialState: { profile: null, profileLoading: true },
      },
    );
    expect(getByTestId("account-container")).toBeTruthy();
    expect(queryByTestId("account-column")).toBeNull();
    expect(queryByAltText("profile image")).toBeNull();
  });

  it("render user data correctly after loading", async () => {
    const afterLoadingState = {
      profile: {
        displayName: "Zabashtevich",
        photoURL:
          "https://firebasestorage.googleapis.com/v0/b/tmdb-api-63318.appspot.com/o/unnamed.jpg?alt=media&token=b405c3ee-ba51-4ffd-bc4f-f1cf91e544ae",
      },
      profileLoading: false,
    };

    const {
      getByText,
      getByTestId,
      getByRole,
      getByAltText,
      container,
    } = renderWithRedux(<AccountContainer />, {
      initialState: {
        userProfile: {
          ...afterLoadingState,
        },
      },
    });
    expect(getByTestId("account-container")).toBeTruthy();
    expect(getByText("Zabashtevich")).toBeTruthy();
    expect(getByRole("img")).toBeTruthy();
    expect(getByAltText("profile image").src).toContain(
      "https://firebasestorage.googleapis.com/v0/b/tmdb-api-63318.appspot.com/o/unnamed.jpg?alt=media&token=b405c3ee-ba51-4ffd-bc4f-f1cf91e544ae",
    );
    expect(getByText(/your profile activity/i)).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
