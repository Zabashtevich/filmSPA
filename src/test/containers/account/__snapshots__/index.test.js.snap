// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account container render user data correctly after loading 1`] = `
<div
  class="account__Container-sc-2uwcdi-0 gUELBw"
  data-testid="account-container"
>
  <div
    class="account__Column-sc-2uwcdi-1 jihhBM"
  >
    <img
      alt="profile image"
      class="account__Poster-sc-2uwcdi-2 istUAj"
      src="https://firebasestorage.googleapis.com/v0/b/tmdb-api-63318.appspot.com/o/unnamed.jpg?alt=media&token=b405c3ee-ba51-4ffd-bc4f-f1cf91e544ae"
    />
    <span
      class="account__Nickname-sc-2uwcdi-3 favWHE"
    >
      Zabashtevich
    </span>
  </div>
  <div
    class="account__Content-sc-2uwcdi-4 jIARfU"
  >
    <span
      class="account__Title-sc-2uwcdi-5 bOFUMs"
    >
      YOUR PROFILE ACTIVITY
    </span>
    <div
      class="filter__Container-sc-1yp3tcu-0 lhaIzW"
    >
      <span
        class="filter__Title-sc-1yp3tcu-1 bCAYEv"
      >
        Customize your activity list
      </span>
      <div
        class="filter__Wrapper-sc-1yp3tcu-2 hsAkBu"
      >
        <div
          class="filter__Row-sc-1yp3tcu-3 ifCEPw"
        >
          <span
            class="filter__Name-sc-1yp3tcu-4 kumIAg"
          >
            Sort by:
          </span>
          <span
            class="filter__Value-sc-1yp3tcu-5 dudyCf"
          >
            DATE
          </span>
          <span
            class="filter__Value-sc-1yp3tcu-5 dudyCf"
          >
            POPULARITY
          </span>
          <span
            class="filter__Value-sc-1yp3tcu-5 dudyCf"
          >
            YOUR VOTE VALUE
          </span>
          <span
            class="filter__Value-sc-1yp3tcu-5 dudyCf"
          >
            AMOUNT OF VOTES
          </span>
          <span
            class="filter__Value-sc-1yp3tcu-5 dudyCf"
          >
            VOTE AVERAGE
          </span>
        </div>
        <div
          class="filter__Row-sc-1yp3tcu-3 ifCEPw"
        >
          <span
            class="filter__Name-sc-1yp3tcu-4 kumIAg"
          >
            List type:
          </span>
          <span
            class="filter__Value-sc-1yp3tcu-5 dudyCf"
          >
            VOTED
          </span>
          <span
            class="filter__Value-sc-1yp3tcu-5 dudyCf"
          >
            USERLIST
          </span>
          <span
            class="filter__Value-sc-1yp3tcu-5 dudyCf"
          >
            FAVORITE
          </span>
        </div>
        <div
          class="filter__Row-sc-1yp3tcu-3 ifCEPw"
        >
          <span
            class="filter__Name-sc-1yp3tcu-4 kumIAg"
          >
            Your list:
          </span>
        </div>
        <div
          class="filter__Row-sc-1yp3tcu-3 ifCEPw"
        >
          <span
            class="filter__Name-sc-1yp3tcu-4 kumIAg"
          >
            Period:
          </span>
          <span
            class="filter__Subtitle-sc-1yp3tcu-6 bugdKQ"
          >
            from
          </span>
          <select
            class="filter__Select-sc-1yp3tcu-7 eJdIRW"
          >
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="all"
            >
              ALL
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1950"
            >
              1950
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1951"
            >
              1951
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1952"
            >
              1952
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1953"
            >
              1953
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1954"
            >
              1954
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1955"
            >
              1955
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1956"
            >
              1956
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1957"
            >
              1957
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1958"
            >
              1958
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1959"
            >
              1959
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1960"
            >
              1960
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1961"
            >
              1961
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1962"
            >
              1962
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1963"
            >
              1963
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1964"
            >
              1964
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1965"
            >
              1965
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1966"
            >
              1966
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1967"
            >
              1967
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1968"
            >
              1968
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1969"
            >
              1969
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1970"
            >
              1970
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1971"
            >
              1971
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1972"
            >
              1972
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1973"
            >
              1973
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1974"
            >
              1974
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1975"
            >
              1975
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1976"
            >
              1976
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1977"
            >
              1977
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1978"
            >
              1978
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1979"
            >
              1979
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1980"
            >
              1980
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1981"
            >
              1981
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1982"
            >
              1982
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1983"
            >
              1983
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1984"
            >
              1984
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1985"
            >
              1985
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1986"
            >
              1986
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1987"
            >
              1987
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1988"
            >
              1988
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1989"
            >
              1989
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1990"
            >
              1990
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1991"
            >
              1991
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1992"
            >
              1992
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1993"
            >
              1993
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1994"
            >
              1994
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1995"
            >
              1995
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1996"
            >
              1996
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1997"
            >
              1997
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1998"
            >
              1998
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1999"
            >
              1999
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2000"
            >
              2000
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2001"
            >
              2001
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2002"
            >
              2002
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2003"
            >
              2003
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2004"
            >
              2004
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2005"
            >
              2005
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2006"
            >
              2006
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2007"
            >
              2007
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2008"
            >
              2008
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2009"
            >
              2009
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2010"
            >
              2010
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2011"
            >
              2011
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2012"
            >
              2012
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2013"
            >
              2013
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2014"
            >
              2014
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2015"
            >
              2015
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2016"
            >
              2016
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2017"
            >
              2017
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2018"
            >
              2018
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2019"
            >
              2019
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2020"
            >
              2020
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2021"
            >
              2021
            </option>
          </select>
          <span
            class="filter__Subtitle-sc-1yp3tcu-6 bugdKQ"
          >
            to
          </span>
          <select
            class="filter__Select-sc-1yp3tcu-7 eJdIRW"
          >
            <option
              class="filter__Option-sc-1yp3tcu-8"
            >
              ALL
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1950"
            >
              1950
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1951"
            >
              1951
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1952"
            >
              1952
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1953"
            >
              1953
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1954"
            >
              1954
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1955"
            >
              1955
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1956"
            >
              1956
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1957"
            >
              1957
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1958"
            >
              1958
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1959"
            >
              1959
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1960"
            >
              1960
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1961"
            >
              1961
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1962"
            >
              1962
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1963"
            >
              1963
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1964"
            >
              1964
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1965"
            >
              1965
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1966"
            >
              1966
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1967"
            >
              1967
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1968"
            >
              1968
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1969"
            >
              1969
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1970"
            >
              1970
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1971"
            >
              1971
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1972"
            >
              1972
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1973"
            >
              1973
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1974"
            >
              1974
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1975"
            >
              1975
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1976"
            >
              1976
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1977"
            >
              1977
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1978"
            >
              1978
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1979"
            >
              1979
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1980"
            >
              1980
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1981"
            >
              1981
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1982"
            >
              1982
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1983"
            >
              1983
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1984"
            >
              1984
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1985"
            >
              1985
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1986"
            >
              1986
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1987"
            >
              1987
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1988"
            >
              1988
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1989"
            >
              1989
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1990"
            >
              1990
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1991"
            >
              1991
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1992"
            >
              1992
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1993"
            >
              1993
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1994"
            >
              1994
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1995"
            >
              1995
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1996"
            >
              1996
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1997"
            >
              1997
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1998"
            >
              1998
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="1999"
            >
              1999
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2000"
            >
              2000
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2001"
            >
              2001
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2002"
            >
              2002
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2003"
            >
              2003
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2004"
            >
              2004
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2005"
            >
              2005
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2006"
            >
              2006
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2007"
            >
              2007
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2008"
            >
              2008
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2009"
            >
              2009
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2010"
            >
              2010
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2011"
            >
              2011
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2012"
            >
              2012
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2013"
            >
              2013
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2014"
            >
              2014
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2015"
            >
              2015
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2016"
            >
              2016
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2017"
            >
              2017
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2018"
            >
              2018
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2019"
            >
              2019
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2020"
            >
              2020
            </option>
            <option
              class="filter__Option-sc-1yp3tcu-8"
              value="2021"
            >
              2021
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</div>
`;
